en/02_Developer_Guides/09_Security/05_Secure_Coding.md

```diff
@@ -660,115 +660,159 @@ This is a recommended option to secure any controller which displays
-SS_TRUSTED_PROXY_IP_HEADER="HTTP_X_FORWARDED_FOR"
-SS_TRUSTED_PROXY_PROTOCOL_HEADER="HTTP_X_FORWARDED_PROTOCOL"
+---
+after: requestprocessors
+---
+SilverStripe\Core\Injector\Injector:
+  SilverStripe\Control\Middleware\TrustedProxyMiddleware:
+    properties:
+      TrustedProxyIPs:
+        - '127.0.0.1'
+        - '192.168.0.1'
 ```
 
-At the same time, you'll also need to define which headers you trust from these proxy IPs. Since there are multiple ways through which proxies can pass through HTTP information on the original hostname, IP and protocol, these values need to be adjusted for your specific proxy. The header names match their equivalent `$_SERVER` values.
-
-If there is no proxy server, 'none' can be used to distrust all clients.
+If there is no proxy server, 'none' can be used to explicitly distrust all clients.
 If only trusted servers will make requests then you can use '*' to trust all clients.
 Otherwise a comma separated list of individual IP addresses (or subnets in CIDR notation) should be declared.
 
-This behaviour is enabled whenever `SS_TRUSTED_PROXY_IPS` is defined, or if the
-`BlockUntrustedIPs` environment variable is declared. It is advisable to include the
-following in your .htaccess to ensure this behaviour is activated.
-
-```text
-<IfModule mod_env.c>
-    # Ensure that X-Forwarded-Host is only allowed to determine the request
-    # hostname for servers ips defined by SS_TRUSTED_PROXY_IPS in your .env
-    # Note that in a future release this setting will be always on.
-    SetEnv BlockUntrustedIPs true
-</IfModule>
-```
+At the same time, you'll also need to define which headers you trust from these proxy IPs. Since there are multiple ways through which proxies can pass through HTTP information on the original hostname, IP and protocol, these values need to be adjusted for your specific proxy. The header names match their equivalent `$_SERVER` values.
 
-This behaviour is on by default; the environment variable is not required. For correct operation, it is necessary to always set `SS_TRUSTED_PROXY_IPS` if using a proxy.
+If you wish to change the headers that are used to find the proxy information, you should reconfigure the
+`TrustedProxyMiddleware` service:
 
-## Secure sessions, cookies and TLS (HTTPS)
+```yml
+SilverStripe\Control\TrustedProxyMiddleware:
+  properties:
+    ProxyHostHeaders: X-Forwarded-Host
+    ProxySchemeHeaders: X-Forwarded-Protocol
+    ProxyIPHeaders: X-Forwarded-Ip
+```
 
-Silverstripe CMS recommends the use of TLS (HTTPS) for your application, and you can easily force the use through the
-director function `forceSSL()`
+## TLS (aka SSL aka HTTPS)
 
-```php
-use SilverStripe\Control\Director;
+Silverstripe CMS recommends the use of TLS (HTTPS) for your application. You can configure this by setting the `ForceSSL` property on the [`CanonicalURLMiddleware`](api:SilverStripe\Control\Middleware\CanonicalURLMiddleware) singleton.
 
-if (!Director::isDev()) {
-    Director::forceSSL();
```

Comment:
This API does exist but it's a lot less powerful than configuring via YAML because
- there are a lot of options on `CanonicalURLMiddleware` that can't be set via director (even the rest of this old docs has to instantiate a singleton of the middleware to set something)
- you ideally want everything configuring this in the same place - so if you have to set some of it in YAML and then some here, that's hard to notice and may cause problems
- basically everywhere that sets `CanonicalURLMiddleware` config uses YAML, following CWP's example.